bulkimport_scripts/add_users_for_bulk_import/index.js

```diff
@@ -0,0 +1,191 @@
+    process.exit(1);
+  }
+}
+
+async function deleteUsersHavingInvalidSchemaFileIfExists({ usersHavingInvalidSchemaFileName }) {
+  try {
+    await fs.rm(usersHavingInvalidSchemaFileName);
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      console.error(`Failed to delete ${usersHavingInvalidSchemaFileName}:`, error.message);
+    }
+  }
+}
+
+async function addInvalidSchemaUsersToFile({ errors, users, usersHavingInvalidSchemaFileName }) {
+  let parsedData = null;
+  try {
+    const existingData = await fs.readFile(usersHavingInvalidSchemaFileName, 'utf8');
+    parsedData = JSON.parse(existingData);
+  } catch (error) {
+    if (error.code === 'ENOENT') {
+      parsedData = { users: [] };
+    } else {
+      console.error(`Failed to read output file. Error: ${error.message}`);
+      throw error;
+    }
+  }
+
+  parsedData.users.push(...errors.map((err) => ({ user: users[err.index], errors: err.errors })));
+
+  await fs.writeFile(usersHavingInvalidSchemaFileName, JSON.stringify(parsedData, null, 2));
+
+  return users.filter((_, index) => !errors.some(err => err.index === index));
+}
+
+async function updateRemainingUsersFile({ users, index, remainingUsersFileName }) {
+  const remainingUsers = users.slice(index + 1);
+  await fs.writeFile(remainingUsersFileName, JSON.stringify({ users: remainingUsers }, null, 2));
+}
+
+async function removeRemainingUsersFile({ remainingUsersFileName }) {
+  try {
+    await fs.rm(remainingUsersFileName);
+  } catch (error) {
+    if (error.code !== 'ENOENT') {
+      console.error(`Failed to delete ${remainingUsersFileName}:`, error.message);
+    }
+  }
+}
+
+async function main() {
+  const { coreAPIUrl, inputFileName, usersHavingInvalidSchemaFileName, remainingUsersFileName } = await parseInputArgs();
+
+  const users = await getUsersFromInputFile({ inputFileName });
+
+  await deleteUsersHavingInvalidSchemaFileIfExists({ usersHavingInvalidSchemaFileName });
+  await updateRemainingUsersFile({ users,  index: 0, remainingUsersFileName });
+
+  let usersToProcessInBatch = [];
+  let usersHavingInvalidSchemaCount = 0;
+  let i = 0;
+
+  try {
+    while (i < users.length || usersToProcessInBatch.length > 0) {
+      let remainingBatchSize = usersToProcessInBatch.length > BATCH_SIZE ? 0 : BATCH_SIZE - usersToProcessInBatch.length;
+      remainingBatchSize = Math.min(remainingBatchSize, users.length - i);
+
+      usersToProcessInBatch.push(...users.slice(i, i + remainingBatchSize));
+
+      const res = await fetch(`${coreAPIUrl}/bulk-import/users`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ users: usersToProcessInBatch }),
+      });
+
+      if (!res.ok && res.status !== 400) {
+        const text = await res.text();
+        console.error(`Failed to add users. API response - status: ${res.status} body: ${text}`);
+        break;
```

Comment:
not a good idea to break. Imaging I am running this script for many 100k users, and went away for a coffee. Then after 5 seconds this fails temporarily. Now it will make no progress even if the core is up. Instead, do exponential backoff (upto a few seconds max), and try again.